src/Akeneo/UserManagement/Bundle/Security/JobUserProvider.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\UserManagement\Bundle\Security;
+
+use Akeneo\UserManagement\Component\Repository\UserRepositoryInterface;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\Exception\UserNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class JobUserProvider implements UserProviderInterface
+{
+    public function __construct(
+        private UserRepositoryInterface $userRepository,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadUserByUsername($username)
+    {
+        $user = $this->userRepository->findOneByIdentifier($username);
+        if (!$user || $user->isApiUser()) {
+            throw new UserNotFoundException(sprintf('User with username "%s" does not exist.', $username));
```

Comment:
This exception text could be misleading I think if the use really exists but is an API User. 
```suggestion
            throw new UserNotFoundException(sprintf('Job user with username "%s" does not exist, or exists but as an API user.', $username));
```